### PR TITLE
🐛 Fixed packages mypy, pylint and tests

### PR DESCRIPTION
**Pull request by GitHK**
## What do these changes do?

Made packages tests green.

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

File: packages/models-library/src/models_library/api_schemas_directorv2/comp_tasks.py [hunk 1]
```diff
@@ -18,6 +18,7 @@ class ComputationGet(ComputationTask):
     stop_url: AnyHttpUrl | None = Field(
         None, description="the link where to stop the task"
     )
+    model_config = ConfigDict(json_schema_extra={"examples": []})
```
Comment (sanderegg) [inline packages/models-library/src/models_library/api_schemas_directorv2/comp_tasks.py:21]: this looks wrong

File: packages/models-library/src/models_library/api_schemas_dynamic_sidecar/telemetry.py [hunk 2]
```diff
@@ -56,29 +55,13 @@ class DiskUsage(BaseModel):
     free: ByteSize = Field(description="remaining space")
 
     total: ByteSize = Field(description="total space = free + used")
-    used_percent: NonNegativeFloat = Field(
+    used_percent: float = Field(
         ge=0.00,
         le=100.00,
-        description="Percent of used space relative to the total space",
+        description="Percent of used space ree:ative to the total space",
```
Comment (sanderegg) [inline packages/models-library/src/models_library/api_schemas_dynamic_sidecar/telemetry.py:61]: did you use AI to correct that one? that looks wrong

File: packages/models-library/src/models_library/api_schemas_dynamic_sidecar/telemetry.py [hunk 3]
```diff
@@ -56,29 +55,13 @@ class DiskUsage(BaseModel):
     free: ByteSize = Field(description="remaining space")
 
     total: ByteSize = Field(description="total space = free + used")
-    used_percent: NonNegativeFloat = Field(
+    used_percent: float = Field(
```
Comment (sanderegg) [inline packages/models-library/src/models_library/api_schemas_dynamic_sidecar/telemetry.py:58]: is nonnegativefloat not anymore existing?

File: packages/models-library/src/models_library/api_schemas_webserver/projects.py [hunk 4]
```diff
@@ -83,11 +83,11 @@ class ProjectGet(OutputSchema):
...
     )
-    dev: dict | None
+    dev: dict | None = None
     permalink: ProjectPermalink | None = None
-    workspace_id: WorkspaceID | None
-    folder_id: FolderID | None
-    trashed_at: datetime | None
+    workspace_id: WorkspaceID | None = None
+    folder_id: FolderID | None = None
+    trashed_at: datetime | None = None
```
Comment (sanderegg) [inline packages/models-library/src/models_library/api_schemas_webserver/projects.py:91]: thes changes look suspicious. this is an output and you made them now optional which they were not. can you double check please?

File: packages/models-library/src/models_library/api_schemas_storage.py [hunk 5]
```diff
@@ -317,6 +317,7 @@ class FoldersBody(BaseModel):
 
     @model_validator(mode="after")
     def ensure_consistent_entries(self) -> Self:
+        # pylint:disable=no-member
```
Comment (sanderegg) [inline packages/models-library/src/models_library/api_schemas_storage.py:320]: why do you need this here?

File: packages/models-library/src/models_library/progress_bar.py [hunk 6]
```diff
@@ -12,8 +12,8 @@ class ProgressStructuredMessage(BaseModel):
     description: IDStr
     current: float
     total: int
-    unit: str | None
-    sub: "ProgressStructuredMessage | None"
+    unit: str | None = None
```
Comment (sanderegg) [inline packages/models-library/src/models_library/progress_bar.py:15]: I don't think that was the intention.
I think the tests are probably the problem. Please don't set things as optional when that is not the intention.

File: packages/models-library/src/models_library/rest_pagination.py [hunk 7]
```diff
@@ -136,34 +136,4 @@ def _check_data_compatible_with_meta(cls, v, info: ValidationInfo):
...
-                        "prev": "https://osparc.io/v2/listing?offset=0&limit=4",
-                        "next": None,
-                        "last": "https://osparc.io/v2/listing?offset=1&limit=4",
-                    },
-                    "data": ["data 5", "data 6", "data 7"],
-                },
-            ]
-        },
-    )
+    model_config = ConfigDict(extra="forbid")
```
Comment (sanderegg) [inline packages/models-library/src/models_library/rest_pagination.py:139]: hmm, the whole example goes away??

File: packages/models-library/src/models_library/rpc_pagination.py [hunk 8]
```diff
@@ -74,34 +74,4 @@ def create(
...
-                        "prev": {"offset": 0, "limit": 4},
-                        "next": None,
-                        "last": {"offset": 1, "limit": 4},
-                    },
-                    "data": ["data 5", "data 6", "data 7"],
-                },
-            ]
-        },
-    )
+    model_config = ConfigDict(extra="forbid")
```
Comment (sanderegg) [inline packages/models-library/src/models_library/rpc_pagination.py:77]: same here???

File: packages/models-library/src/models_library/service_settings_labels.py [hunk 9]
```diff
@@ -456,6 +456,7 @@ def _user_preferences_path_no_included_in_other_volumes(
 
     @model_validator(mode="after")
     def _not_allowed_in_both_specs(self):
+        # pylint:disable=no-member
```
Comment (sanderegg) [inline packages/models-library/src/models_library/service_settings_labels.py:459]: did you try to set `self: Self` ?

File: packages/simcore-sdk/src/simcore_sdk/node_ports_common/dbmanager.py [hunk 10]
```diff
@@ -73,6 +73,7 @@ async def _create_db_engine() -> aiopg.sa.Engine:
         settings = NodePortsSettings.create_from_envs()
         dsn = DataSourceName(
             application_name=f"{__name__}_{socket.gethostname()}_{os.getpid()}",
+            # pylint:disable=no-member
```
Comment (sanderegg) [inline packages/simcore-sdk/src/simcore_sdk/node_ports_common/dbmanager.py:76]: this looks weird

File: packages/simcore-sdk/src/simcore_sdk/node_ports_common/storage_endpoint.py [hunk 11]
```diff
@@ -8,13 +8,15 @@
 def is_storage_secure() -> bool:
     settings = NodePortsSettings.create_from_envs()
     node_ports_storage_auth = settings.NODE_PORTS_STORAGE_AUTH
+    # pylint:disable=no-member
```
Comment (sanderegg) [inline packages/simcore-sdk/src/simcore_sdk/node_ports_common/storage_endpoint.py:11]: same here

**Review by sanderegg — COMMENTED** [review]
there are a few things that are strange. Also setting models to have optional fields has consequences. I would like to check the one with the progress bar at least

File: packages/models-library/src/models_library/api_schemas_storage.py (hunk 5 above)
Comment (GitHK) [reply packages/models-library/src/models_library/api_schemas_storage.py]: correct with source: Annotated[dict[str, Any], Field(default_factory=dict)]
all of them

File: packages/models-library/src/models_library/progress_bar.py (hunk 6 above)
Comment (GitHK) [reply packages/models-library/src/models_library/progress_bar.py:15]: this is ok

File: packages/models-library/src/models_library/api_schemas_directorv2/comp_tasks.py (hunk 1 above)
Comment (pcrespov) [reply packages/models-library/src/models_library/api_schemas_directorv2/comp_tasks.py:21]: Q: why empty example? Either remove it or add an example, right?

File: packages/models-library/src/models_library/api_schemas_dynamic_sidecar/telemetry.py (hunk 3 above)
Comment (pcrespov) [reply packages/models-library/src/models_library/api_schemas_dynamic_sidecar/telemetry.py:58]: I guess you guys talked about this  but not sure what the outcome was :-)

File: packages/models-library/src/models_library/rpc_pagination.py [hunk 12]
```diff
@@ -74,34 +74,4 @@ def create(
             data=chunk,
         )
 
-    model_config = ConfigDict(
```
Comment (pcrespov) [inline packages/models-library/src/models_library/rpc_pagination.py:77]: OK, so as we discussed, we cannot use the examples in generics as before, right?

File: packages/models-library/tests/shared_examples.py
Comment (pcrespov) [inline packages/models-library/tests/shared_examples.py:1]: We found sometime ago that the best way to have helpers for testers is to have the in `pytest_simcore`
This way we avoid other issues e.g. with import or creating a dependnecy with the cwd of in which pytest is run ... etc

**Review by pcrespov — APPROVED** [review]

**Review by giancarloromeo — APPROVED** [review]

File: packages/models-library/src/models_library/api_schemas_directorv2/clusters.py [hunk 13]
```diff
@@ -113,7 +113,7 @@ class ClusterDetailsGet(ClusterDetails):
 
 
 class ClusterCreate(BaseCluster):
-    owner: GroupID | None
+    owner: GroupID | None = None  # type: ignore[assignment]
```
Comment (sanderegg) [inline packages/models-library/src/models_library/api_schemas_directorv2/clusters.py:116]: actually I did not see the `= None` in my last check. only the ignore.
why is this changed to optional? another test?

File: packages/models-library/src/models_library/api_schemas_directorv2/comp_tasks.py (hunk 1 above)
Comment (sanderegg) [reply packages/models-library/src/models_library/api_schemas_directorv2/comp_tasks.py:21]: did you forget that one?

**Review by sanderegg — APPROVED** [review]
Please check ClusterCreate. thanks for the other changes.

File: packages/models-library/src/models_library/api_schemas_directorv2/clusters.py (hunk 13 above)
Comment (GitHK) [reply packages/models-library/src/models_library/api_schemas_directorv2/clusters.py:116]: because of the below attached example. One has the owner field one does not have it expiclty specified. So it implies it has to be None by default

File: packages/models-library/src/models_library/api_schemas_directorv2/comp_tasks.py (hunk 1 above)
Comment (GitHK) [reply packages/models-library/src/models_library/api_schemas_directorv2/comp_tasks.py]: Forgot to extend the examples here
